tests/unit/Storefront/Framework/Seo/SeoUrlRoute/NavigationPageSeoUrlRouteTest.php

```diff
@@ -34,29 +34,47 @@ public function testPrepareCriteria(): void
         $navigationPageSeoUrlRoute->prepareCriteria($criteria, $salesChannel);
 
         $filters = $criteria->getFilters();
-        static::assertCount(2, $filters);
+        /** @var MultiFilter $multiFilter */
+        $multiFilter = $filters[0];
+        static::assertInstanceOf(MultiFilter::class, $multiFilter);
+        static::assertEquals('AND', $multiFilter->getOperator());
+        $multiFilterQueries = $multiFilter->getQueries();
 
-        $notFilter = $filters[0];
-        static::assertInstanceOf(NotFilter::class, $notFilter);
+        static::assertCount(2, $multiFilterQueries);
+        static::assertInstanceOf(EqualsFilter::class, $multiFilterQueries[0]);
+        $this->assertEqualsFilter(
+            $multiFilterQueries[0],
+            'active',
+            true
+        );
 
-        static::assertEquals(MultiFilter::CONNECTION_OR, $notFilter->getOperator());
+        $notFilter = $multiFilterQueries[1];
+        static::assertInstanceOf(NotFilter::class, $notFilter);
+        static::assertEquals('OR', $notFilter->getOperator());
 
         $notFilterQueries = $notFilter->getQueries();
         static::assertCount(2, $notFilterQueries);
+        static::assertInstanceOf(EqualsFilter::class, $notFilterQueries[0]);
+        $this->assertEqualsFilter(
+            $notFilterQueries[0],
+            'type',
+            'folder'
+        );
+        static::assertInstanceOf(EqualsFilter::class, $notFilterQueries[1]);
+        $this->assertEqualsFilter(
+            $notFilterQueries[1],
+            'type',
+            'link'
+        );
+    }
 
-        $equalsFilter = $notFilterQueries[0];
+    private function assertEqualsFilter(
+        EqualsFilter $equalsFilter,
+        string $field,
+        string|bool $value
+    ): void {
         static::assertInstanceOf(EqualsFilter::class, $equalsFilter);
```

Comment:
You do this assert twice, once before you call this method and then here again, also this check is redundant as the type is already checked in the method signature

but i would remove the assertInstanceOf checks in the tests itself and declare the parameter type as `object` here, so you only need to do the instance of check once ;) 

BTW this should also solve the phpstan error that you added to the baseline